src/components/Board.js

```diff
@@ -5,29 +5,103 @@ import axios from 'axios';
 import './Board.css';
 import Card from './Card';
 import NewCardForm from './NewCardForm';
-import CARD_DATA from '../data/card-data.json';
 
 class Board extends Component {
   constructor() {
     super();
 
     this.state = {
       cards: [],
+      errorMessages: []
     };
   }
 
+  componentDidMount() {
+
+    const GET_ALL_CARDS_URL = `https://inspiration-board.herokuapp.com/boards/${this.props.boardName}/cards`;
+
+    axios.get(GET_ALL_CARDS_URL)
+    .then((response) => {
+      this.setState({ cards: response.data });
+    })
+    .catch((error) => {
+      this.setState({
+        errorMessages: [...this.state.errorMessages, error.message]
+      });
+    });
+  }
+
+  addCard = (cardData) => {
+
+    const ADD_CARD_URL = `https://inspiration-board.herokuapp.com/boards/${this.props.boardName}/cards`;
+
+    axios.post(ADD_CARD_URL, cardData)
+    .then((response) => {
+      this.setState({
+        cards: [ ...this.state.cards, response.data]
+      });
+    })
+    .catch((error) => {
+      this.setState({
+        errorMessages: [...this.state.errorMessages, error.message]
+      });
+    });
+  }
+
+  deleteCard = (id) => {
+    const DELETE_CARD_URL = `https://inspiration-board.herokuapp.com/cards/${id}`;
+
+    axios.delete(DELETE_CARD_URL)
+    .then(() => {
+      const updatedCardList = [...this.state.cards];
+      const index = updatedCardList.findIndex(card => card.id === id);
+      updatedCardList.splice(index, 1);
+
+      this.setState({ cards: updatedCardList });
+    })
+    .catch((error) => {
+      this.setState({ errorMessages: [...this.state.errorMessages, error.message] });
+    });
+  }
+
   render() {
+
+    const cardList = this.state.cards.map((card, i) => {
+
+      const { id, text, emoji } = card.card;
+
+      const formattedCard = {
+        id: id,
+        text: text,
+        emoji: emoji
+      };
+
+      return <Card key={i}
+        card={formattedCard}
+        deleteCard={() => this.deleteCard(id)} />
```

Comment:
Instead of doing the restructuring and working with `formattedCard` here, you could say `card={...card.card}` on line 80.